src/_base/application/overlay/Jenkinsfile.twig

```diff
@@ -22,10 +22,20 @@ pipeline {
             steps { sh 'ws app publish' }
         }
 {% endif %}
-{% if @('pipeline.preview.enabled') == 'yes' %}
-        stage('Deploy') {
-            when { not { triggeredBy 'TimerTrigger' } }
-            steps { sh 'ws app deploy preview' }
+{% if @('pipeline.qa.enabled') == 'yes' %}
+        stage('Deploy (QA)') {
+            environment {
+{% for key, value in @('pipeline.qa.environment') %}
+                {{ key }} = {{ value|raw }}
+{% endfor %}
```

Comment:
The "qa" environment in helm is having values.yml file which contains all these environment variables, do we still need to pass them here?

Reply:
perhaps the naming needs to be improved, this is to allow the jenkins agent access to values eg.

`  pipeline:
    publish:
      enabled: yes
    qa:
      enabled: yes
      environment:
        DO_ACCESS_TOKEN: credentials('inviqa-digitalocean-token')
      cluster:
        name: inviqa-playground
      namespace: cms-qa
      hostname:  cms-qa.playground.webpipeline.net`